aws/resource_aws_chime_voice_connector_termination.go

```diff
@@ -0,0 +1,191 @@
+		VoiceConnectorId: aws.String(vcId),
+	}
+
+	resp, err := conn.GetVoiceConnectorTerminationWithContext(ctx, input)
+	if !d.IsNewResource() && isAWSErr(err, chime.ErrCodeNotFoundException, "") {
+		log.Printf("[WARN] error getting Chime Voice Connector (%s) termination: %s", vcId, err)
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil || resp.Termination == nil {
+		return diag.Errorf("error getting Chime Voice Connector (%s) termination: %s", vcId, err)
+	}
+
+	d.Set("cps_limit", resp.Termination.CpsLimit)
+	d.Set("disabled", resp.Termination.Disabled)
+	d.Set("default_phone_number", resp.Termination.DefaultPhoneNumber)
+
+	if err := d.Set("calling_regions", flattenStringList(resp.Termination.CallingRegions)); err != nil {
+		return diag.Errorf("error setting termination calling regions (%s): %s", vcId, err)
+	}
+	if err := d.Set("cidr_allow_list", flattenStringList(resp.Termination.CidrAllowedList)); err != nil {
+		return diag.Errorf("error setting termination cidr allow list (%s): %s", vcId, err)
+	}
+
+	return nil
+}
+
+func resourceAwsChimeVoiceConnectorTerminationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).chimeconn
+
+	if d.HasChanges("calling_regions", "cidr_allow_list", "disabled", "cps_limit", "default_phone_number") {
+		vcId := d.Get("voice_connector_id").(string)
+		termination := &chime.Termination{
+			CallingRegions:  expandStringList(d.Get("calling_regions").([]interface{})),
+			CidrAllowedList: expandStringList(d.Get("cidr_allow_list").([]interface{})),
+			CpsLimit:        aws.Int64(int64(d.Get("cps_limit").(int))),
+		}
+
+		if v, ok := d.GetOk("default_phone_number"); ok {
+			termination.DefaultPhoneNumber = aws.String(v.(string))
+		}
+
+		if v, ok := d.GetOk("disabled"); ok {
+			termination.Disabled = aws.Bool(v.(bool))
+		}
+
+		input := &chime.PutVoiceConnectorTerminationInput{
+			VoiceConnectorId: aws.String(vcId),
+			Termination:      termination,
+		}
+
+		if _, err := conn.PutVoiceConnectorTerminationWithContext(ctx, input); err != nil {
+			if isAWSErr(err, chime.ErrCodeNotFoundException, "") {
+				log.Printf("[WARN] error getting Chime Voice Connector (%s) termination: %s", vcId, err)
+				d.SetId("")
+				return nil
+			}
+
+			return diag.Errorf("error updating Chime Voice Connector (%s) termination: %s", vcId, err)
+		}
+	}
+
+	return resourceAwsChimeVoiceConnectorTerminationRead(ctx, d, meta)
+}
+
+func resourceAwsChimeVoiceConnectorTerminationDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).chimeconn
+
+	vcId := d.Get("voice_connector_id").(string)
+	input := &chime.DeleteVoiceConnectorTerminationInput{
+		VoiceConnectorId: aws.String(vcId),
+	}
+
+	if _, err := conn.DeleteVoiceConnectorTerminationWithContext(ctx, input); err != nil {
+		return diag.Errorf("error deleting Chime Voice Connector (%s) termination (%s): %s", vcId, d.Id(), err)
+	}
```

Comment:
Can use ID here instead (after changing it above):

```suggestion
	input := &chime.DeleteVoiceConnectorTerminationInput{
		VoiceConnectorId: aws.String(d.Id()),
	}

	_, err := conn.DeleteVoiceConnectorTerminationWithContext(ctx, input)
	
	if isAWSErr(err, chime.ErrCodeNotFoundException, "") {
	        return nil
	}
	
	if err != nil {
		return diag.Errorf("error deleting Chime Voice Connector Termination (%s): %s", d.Id(), err)
	}
```

Reply:
@anGie44 In this case d.Id() is an identifier for termination setting we set at line 102, but we need voice connector id

Reply:
@aleks1001 -- the ID can be set to the `voice_connector_id` since that's data we need to perform CRUD ops 👍